staging/src/k8s.io/component-base/featuregate/feature_gate.go

```diff
@@ -82,6 +82,39 @@ type FeatureSpec struct {
 	// If multiple FeatureSpecs exist for a Feature, the one with the highest version that is less
 	// than or equal to the effective version of the component is used.
 	Version *version.Version
+	// MinCompatibilityVersion indicates the lowest control plane version that this feature spec is compatible with.
+	// If a server's --min-compatibility-version is less than this value, this feature spec entry will be skipped.
+	// This is useful for features with compatibility implications (like a new API field),
+	// allowing them to be introduced as Beta but only on by default when all control plane components are upgraded.
```

Comment:
We *happen* to only be planning to make min-compatibility version configurable for control plane components, but there's nothing *inherently* control-plane specific about this construct.

Since this lives in component-base, let's make the documentation more neutral:

```suggestion
	// MinCompatibilityVersion indicates the lowest version that this feature spec is compatible with.
	// The component's minimum compatibility version must be greater than or equal to this version for this spec to be used.
	// This allows features with skew compatibility implications to be introduced as Beta,
	// but only on by default once the minimum compatibility version is high enough.
```